quran/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import generics
+from rest_framework.response import Response
+from rest_framework.views import status
+
+from . import serializers
+from .models import Surah
+
+
```

Comment:
use this `class SurahView(GenericAPIView, RetrieveModelMixin, ListModelMixin):`. So instead of two views you may have only one.

look at the ListApiView   it inherits from GenericAPIView and ListModelMixin
look at then SurahDetailsView  it inherits from  GenericAPIView and [RetrieveModelMixin](https://www.django-rest-framework.org/api-guide/generic-views/#retrievemodelmixin)

so